<a href="https://colab.research.google.com/github/kavrit/EDA-Airbnb-booking-analsysis-Capstone-project-1/blob/main/EDA_Airbnb_booking_analysis_capstone_project_1_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### <b> Since 2008, guests and hosts have used Airbnb to expand on traveling possibilities and present a more unique, personalized way of experiencing the world. Today, Airbnb became one of a kind service that is used and recognized by the whole world. Data analysis on millions of listings provided through Airbnb is a crucial factor for the company. These millions of listings generate a lot of data - data that can be analyzed and used for security, business decisions, understanding of customers' and providers' (hosts) behavior and performance on the platform, guiding marketing initiatives, implementation of innovative additional services and much more. </b>

## <b>This dataset has around 49,000 observations in it with 16 columns and it is a mix between categorical and numeric values. </b>

## <b> Explore and analyze the data to discover key understandings (not limited to these) such as : 
* What can we learn about different hosts and areas?
* What can we learn from predictions? (ex: locations, prices, reviews, etc)
* Which hosts are the busiest and why?
* Is there any noticeable difference of traffic among different areas and what could be the reason for it? </b>
* **Depending upon the booking, can we segregate the neighborhood among various tires?**
* **What is the exact postal address for the best reviewed hotel?**
* **Which room has the lowest reviews?**
* **Relationship between number of reviews and price ?**



## **Importing required liabraries for Airbnb NYC analysis**

In [1]:
import pandas as pd # for data handling and manipulation
import matplotlib.pyplot as plt # for data visualization
import seaborn as sns # for data visualization
# magic function to render graphs/charts in notebook
%matplotlib inline  

In [2]:
#set max columns display limit
pd.set_option("display.max_columns", 35)
#set style for graph
sns.set_style(style='darkgrid')

# Setting fontsizes,fontweight,label weight for lebels,titles.
plt.rcParams["font.weight"] = "bold"
plt.rcParams["axes.labelweight"] = "bold"
plt.rcParams["axes.titlesize"] = 20
plt.rcParams["axes.titleweight"] = 'bold'
plt.rcParams['xtick.labelsize']=15
plt.rcParams['ytick.labelsize']=15
plt.rcParams["axes.labelsize"] = 20
plt.rcParams["legend.fontsize"] = 15
plt.rcParams["legend.title_fontsize"] = 15

## **Mounting Drive**

In [3]:
from pydrive.auth import GoogleAuth
from pydrive.drive import GoogleDrive
from google.colab import auth
from oauth2client.client import GoogleCredentials
auth.authenticate_user()
gauth = GoogleAuth()
gauth.credentials = GoogleCredentials.get_application_default()
drive = GoogleDrive(gauth)
link='https://drive.google.com/file/d/1ioU5r9KEYSfwgfUi22SclVkx4l1a_8ou/view?usp=sharing'

 
# to get the id part of the file
id = link.split("/")[-2]
 
downloaded = drive.CreateFile({'id':id})
downloaded.GetContentFile('Airbnb NYC 2019.csv') 
 
df = pd.read_csv('Airbnb NYC 2019.csv')

## **Reading data in colab notebook**

In [4]:
#checking top 5 rows of dataset
Airbnb_NYC_df=df
Airbnb_NYC_df.head()

id                                              name  host_id  \
0  2539                Clean & quiet apt home by the park     2787   
1  2595                             Skylit Midtown Castle     2845   
2  3647               THE VILLAGE OF HARLEM....NEW YORK !     4632   
3  3831                   Cozy Entire Floor of Brownstone     4869   
4  5022  Entire Apt: Spacious Studio/Loft by central park     7192   

     host_name neighbourhood_group neighbourhood  latitude  longitude  \
0         John            Brooklyn    Kensington  40.64749  -73.97237   
1     Jennifer           Manhattan       Midtown  40.75362  -73.98377   
2    Elisabeth           Manhattan        Harlem  40.80902  -73.94190   
3  LisaRoxanne            Brooklyn  Clinton Hill  40.68514  -73.95976   
4        Laura           Manhattan   East Harlem  40.79851  -73.94399   

         room_type  price  minimum_nights  number_of_reviews last_review  \
0     Private room    149               1                  9  2018-10-19   
1  Entire home/apt    225               1                 45  2019-05-21   
2     Private room    150               3                  0         NaN   
3  Entire home/apt     89               1                270  2019-07-05   
4  Entire home/apt     80              10                  9  2018-11-19   

   reviews_per_month  calculated_host_listings_count  availability_365  
0               0.21                               6               365  
1               0.38                               2               355  
2                NaN                               1               365  
3               4.64                               1               194  
4               0.10                               1                 0

In [5]:
#checking bottom 5 rows of dataset
Airbnb_NYC_df.tail()

id                                               name   host_id  \
48890  36484665    Charming one bedroom - newly renovated rowhouse   8232441   
48891  36485057      Affordable room in Bushwick/East Williamsburg   6570630   
48892  36485431            Sunny Studio at Historical Neighborhood  23492952   
48893  36485609               43rd St. Time Square-cozy single bed  30985759   
48894  36487245  Trendy duplex in the very heart of Hell's Kitchen  68119814   

           host_name neighbourhood_group       neighbourhood  latitude  \
48890        Sabrina            Brooklyn  Bedford-Stuyvesant  40.67853   
48891        Marisol            Brooklyn            Bushwick  40.70184   
48892  Ilgar & Aysel           Manhattan              Harlem  40.81475   
48893            Taz           Manhattan      Hell's Kitchen  40.75751   
48894     Christophe           Manhattan      Hell's Kitchen  40.76404   

       longitude        room_type  price  minimum_nights  number_of_reviews  \
48890  -73.94995     Private room     70               2                  0   
48891  -73.93317     Private room     40               4                  0   
48892  -73.94867  Entire home/apt    115              10                  0   
48893  -73.99112      Shared room     55               1                  0   
48894  -73.98933     Private room     90               7                  0   

      last_review  reviews_per_month  calculated_host_listings_count  \
48890         NaN                NaN                               2   
48891         NaN                NaN                               2   
48892         NaN                NaN                               1   
48893         NaN                NaN                               6   
48894         NaN                NaN                               1   

       availability_365  
48890                 9  
48891                36  
48892                27  
48893                 2  
48894                23

In [6]:
#Now check information about dataset
Airbnb_NYC_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              48895 non-null  int64  
 1   name                            48879 non-null  object 
 2   host_id                         48895 non-null  int64  
 3   host_name                       48874 non-null  object 
 4   neighbourhood_group             48895 non-null  object 
 5   neighbourhood                   48895 non-null  object 
 6   latitude                        48895 non-null  float64
 7   longitude                       48895 non-null  float64
 8   room_type                       48895 non-null  object 
 9   price                           48895 non-null  int64  
 10  minimum_nights                  48895 non-null  int64  
 11  number_of_reviews               48895 non-null  int64  
 12  last_review                     

Inferences.

1) There are total of 48894 entries.

2) Total of 16 columns.

3) There are Null values in number_of_columns

In [7]:
#now check the summary statistics of numerical columns
Airbnb_NYC_df.describe()

id       host_id      latitude     longitude         price  \
count  4.889500e+04  4.889500e+04  48895.000000  48895.000000  48895.000000   
mean   1.901714e+07  6.762001e+07     40.728949    -73.952170    152.720687   
std    1.098311e+07  7.861097e+07      0.054530      0.046157    240.154170   
min    2.539000e+03  2.438000e+03     40.499790    -74.244420      0.000000   
25%    9.471945e+06  7.822033e+06     40.690100    -73.983070     69.000000   
50%    1.967728e+07  3.079382e+07     40.723070    -73.955680    106.000000   
75%    2.915218e+07  1.074344e+08     40.763115    -73.936275    175.000000   
max    3.648724e+07  2.743213e+08     40.913060    -73.712990  10000.000000   

       minimum_nights  number_of_reviews  reviews_per_month  \
count    48895.000000       48895.000000       38843.000000   
mean         7.029962          23.274466           1.373221   
std         20.510550          44.550582           1.680442   
min          1.000000           0.000000           0.010000   
25%          1.000000           1.000000           0.190000   
50%          3.000000           5.000000           0.720000   
75%          5.000000          24.000000           2.020000   
max       1250.000000         629.000000          58.500000   

       calculated_host_listings_count  availability_365  
count                    48895.000000      48895.000000  
mean                         7.143982        112.781327  
std                         32.952519        131.622289  
min                          1.000000          0.000000  
25%                          1.000000          0.000000  
50%                          1.000000         45.000000  
75%                          2.000000        227.000000  
max                        327.000000        365.000000

In [8]:
#check the column names 
df.columns

Index(['id', 'name', 'host_id', 'host_name', 'neighbourhood_group',
       'neighbourhood', 'latitude', 'longitude', 'room_type', 'price',
       'minimum_nights', 'number_of_reviews', 'last_review',
       'reviews_per_month', 'calculated_host_listings_count',
       'availability_365'],
      dtype='object')

# **Check columns and their unique values to understand what they contain**

In [9]:
#create dataframe of numerical columns
numerical_df = df._get_numeric_data()

In [10]:
#create for loop to print unique values in numerical columns

for col in numerical_df:
  print(f'Unique values in numerical columns {col} are: {numerical_df[col].unique()[:5]}')

Unique values in numerical columns id are: [2539 2595 3647 3831 5022]
Unique values in numerical columns host_id are: [2787 2845 4632 4869 7192]
Unique values in numerical columns latitude are: [40.64749 40.75362 40.80902 40.68514 40.79851]
Unique values in numerical columns longitude are: [-73.97237 -73.98377 -73.9419  -73.95976 -73.94399]
Unique values in numerical columns price are: [149 225 150  89  80]
Unique values in numerical columns minimum_nights are: [ 1  3 10 45  2]
Unique values in numerical columns number_of_reviews are: [  9  45   0 270  74]
Unique values in numerical columns reviews_per_month are: [0.21 0.38  nan 4.64 0.1 ]
Unique values in numerical columns calculated_host_listings_count are: [6 2 1 4 3]
Unique values in numerical columns availability_365 are: [365 355 194   0 129]


In [11]:
#create categorical column dataframe
categorical_columns = df.select_dtypes(include=['object'])
categorical_columns.head()

name    host_name  \
0                Clean & quiet apt home by the park         John   
1                             Skylit Midtown Castle     Jennifer   
2               THE VILLAGE OF HARLEM....NEW YORK !    Elisabeth   
3                   Cozy Entire Floor of Brownstone  LisaRoxanne   
4  Entire Apt: Spacious Studio/Loft by central park        Laura   

  neighbourhood_group neighbourhood        room_type last_review  
0            Brooklyn    Kensington     Private room  2018-10-19  
1           Manhattan       Midtown  Entire home/apt  2019-05-21  
2           Manhattan        Harlem     Private room         NaN  
3            Brooklyn  Clinton Hill  Entire home/apt  2019-07-05  
4           Manhattan   East Harlem  Entire home/apt  2018-11-19

In [12]:
#create for loop to print unique names in categorical columns
for col in categorical_columns:
  print(f'Unique names in categorical columns {col} are: {categorical_columns[col].unique()[:8]}')

Unique names in categorical columns name are: ['Clean & quiet apt home by the park' 'Skylit Midtown Castle'
 'THE VILLAGE OF HARLEM....NEW YORK !' 'Cozy Entire Floor of Brownstone'
 'Entire Apt: Spacious Studio/Loft by central park'
 'Large Cozy 1 BR Apartment In Midtown East' 'BlissArtsSpace!'
 "Large Furnished Room Near B'way "]
Unique names in categorical columns host_name are: ['John' 'Jennifer' 'Elisabeth' 'LisaRoxanne' 'Laura' 'Chris' 'Garon'
 'Shunichi']
Unique names in categorical columns neighbourhood_group are: ['Brooklyn' 'Manhattan' 'Queens' 'Staten Island' 'Bronx']
Unique names in categorical columns neighbourhood are: ['Kensington' 'Midtown' 'Harlem' 'Clinton Hill' 'East Harlem'
 'Murray Hill' 'Bedford-Stuyvesant' "Hell's Kitchen"]
Unique names in categorical columns room_type are: ['Private room' 'Entire home/apt' 'Shared room']
Unique names in categorical columns last_review are: ['2018-10-19' '2019-05-21' nan '2019-07-05' '2018-11-19' '2019-06-22'
 '2017-10-05' '2019-0

we checked columns for their unique values some also contains 0 and Null values.

# **Data Cleaning**

Data cleaning is important step in any EDA process as well as it is most time 
consuimg process in entire EDA.

Data cleaning includes following steps.



1.   Handling missing values.
2.   Removing duplicate data.




## **Step 1 : Handling missing values**

In [13]:
#check null values
df.isnull().sum().sort_values(ascending=False)[:5]

last_review          10052
reviews_per_month    10052
host_name               21
name                    16
id                       0
dtype: int64

## **Inferences**



> There are four columns which contain null values name has very less null values (16), but in last review,review_per_month and host_name we have lots of null values. We have to find a way to replace those null values with something meaningfull.



In [14]:
#replace null values of company agent and children with 0
df[['last_review ','reviews_per_month']] = df[['last_review','reviews_per_month']].fillna(0)

Now in last review column we have missing values some people might have not filled data of last review, So we will replace those null values with 'Not Specified'.


